map_view/display-device-location-with-autopan-modes/src/main/java/com/esri/samples/display_device_location_with_autopan_modes/DisplayDeviceLocationWithAutopanModesSample.java

```diff
@@ -0,0 +1,200 @@
+import javafx.scene.control.Label;
+import javafx.scene.layout.Background;
+import javafx.scene.layout.BackgroundFill;
+import javafx.scene.layout.CornerRadii;
+import javafx.scene.layout.StackPane;
+import javafx.scene.layout.VBox;
+import javafx.scene.paint.Paint;
+import javafx.stage.Stage;
+
+import com.esri.arcgisruntime.geometry.Geometry;
+import com.esri.arcgisruntime.geometry.Polyline;
+import com.esri.arcgisruntime.geometry.SpatialReferences;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.location.SimulatedLocationDataSource;
+import com.esri.arcgisruntime.location.SimulationParameters;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.LocationDisplay;
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+import org.apache.commons.io.IOUtils;
+
+public class DisplayDeviceLocationWithAutopanModesSample extends Application {
+
+  private MapView mapView;
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+      // create the stack pane and the application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set a title, size, and add the scene to stage
+      stage.setTitle("Display Device Location With Autopan Modes Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+      scene.getStylesheets().add(getClass().getResource("/display_device_location_with_autopan_modes/style.css").toExternalForm());
+
+      // create an ArcGISMap with the imagery basemap
+      ArcGISMap map = new ArcGISMap(Basemap.createImagery());
+
+      // create a view and add the ArcGISMap to it
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // create a combo box
+      ComboBox<String> comboBox = new ComboBox<>();
+      comboBox.setMaxWidth(Double.MAX_VALUE);
+      comboBox.setDisable(true);
+      // add the autopan modes to the combo box
+      comboBox.getItems().addAll("Recenter", "Navigation", "Compass", "Off");
+      comboBox.setValue("Recenter");
+
+      // add a label
+      Label autopanModeLabel = new Label("Choose an autopan mode:");
+      // add a checkbox that toggles the visibility of the location display
+      CheckBox checkBox = new CheckBox("Show device location");
+      checkBox.setDisable(true);
+
+      // access the json of the location points
+      String polylineData = IOUtils.toString(getClass().getResourceAsStream(
+        "/display_device_location_with_autopan_modes/polyline_data.json"), StandardCharsets.UTF_8);
+      // create a polyline from the location points
+      Polyline locations = (Polyline) Geometry.fromJson(polylineData, SpatialReferences.getWgs84());
+
+      // create a simulated location data source
+      SimulatedLocationDataSource simulatedLocationDataSource = new SimulatedLocationDataSource();
```

Comment:
let's try out `var` again here: there are A LOT of simulatedlocationdatasources here and this is exactly the sort of thing local variable type inference was introduced to help with readability. It's obvious from variable name that it is a simulated location data source so is a good `var` candidate.
 
```suggestion
      var simulatedLocationDataSource = new SimulatedLocationDataSource();
```